cloudinit/util.py

```diff
@@ -393,14 +393,24 @@ def multi_log(
         sys.stderr.write(text)
     if console:
         conpath = "/dev/console"
+        writing_to_console_worked = False
         if os.path.exists(conpath):
-            with open(conpath, "w") as wfh:
-                wfh.write(text)
-                wfh.flush()
-        elif fallback_to_stdout:
-            # A container may lack /dev/console (arguably a container bug).  If
-            # it does not exist, then write output to stdout.  this will result
-            # in duplicate stderr and stdout messages if stderr was True.
+            try:
+                with open(conpath, "w") as wfh:
+                    wfh.write(text)
+                    wfh.flush()
+                writing_to_console_worked = True
+            except OSError:
```

Comment:
so we get into this situation if console is enabled and /dev/console exists, which indicates some sort of issue with the image. I would log a warning here indicate that serial console is likely busted

Reply:
/dev/console being unwritable to is not necessarily a sign that the image is busted (see e.g. https://www.linuxquestions.org/questions/linux-software-2/enable-dev-ttys0-on-centos-7-vm-on-esx-server-4175606689/).

Regardless, I've added a log.